drivers/volume/kdmp/kdmp.go

```diff
@@ -617,6 +617,24 @@ func (k *kdmp) StartRestore(
 		volumeInfo.SourceVolume = bkpvInfo.Volume
 		volumeInfo.DriverName = storkvolume.KDMPDriverName
 
+		// If restore replace policy is retain , then skip the pvc
+		if restore.Spec.ReplacePolicy == storkapi.ApplicationRestoreReplacePolicyRetain {
+			pvc, err := core.Instance().GetPersistentVolumeClaim(pvc.Name, restoreNamespace)
+			if err != nil && !k8serror.IsNotFound(err) {
+				logrus.Errorf("failed to get pvc object for %s/%s: %v", restoreNamespace, pvc.Name, err)
+				volumeInfo.Status = storkapi.ApplicationRestoreStatusFailed
+				volumeInfos = append(volumeInfos, volumeInfo)
+				continue
+			}
+			if err == nil {
+				logrus.Debugf("Skipping pvc %s/%s as the replace policy is set as %s", restoreNamespace, pvc.Name, storkapi.ApplicationRestoreReplacePolicyRetain)
+				volumeInfo.Status = storkapi.ApplicationRestoreStatusSuccessful
```

Comment:
Shouldn't this status be `ApplicationRestoreStatusPartialSuccess` as we have not restored one PVC?
If this is all successful, then the final status in the applicationrestore CR will also be successful and we should actually show `PartialSuccess` here

Reply:
For volumes, it needs to be Successful and in resources will make the status to be retained so that the overall restore becomes partialSuccess.

Reply:
For volumes also , now adding the status as Retained as those are getting used for determining restore status as successful or partialSuccess.